crates/release_plz_core/src/command/release_pr.rs

```diff
@@ -311,9 +311,49 @@ async fn github_force_push(
     pr: &GitPr,
     repository: &Repo,
 ) -> anyhow::Result<()> {
-    // first force push the branch and then add new commit using gql api
-    repository.force_push(pr.branch())?;
-    github_graphql::commit_changes(client, repository, "chore: release", pr.branch()).await
+    // note: see https://github.com/MarcoIeni/release-plz/issues/1487
+    // Create a 'Verified' commit on temporary branch
+    let tmp_release_branch = {
+        let name = format!("{}-tmp-{}", pr.branch(), rand::random::<u32>());
+        TmpBranch::checkout_new(repository, name)
+    }?;
+
+    github_create_release_branch(client, repository, &tmp_release_branch.name).await?;
+    // rewrite the PR branch to point to the new commit
+    repository.fetch(&tmp_release_branch.name)?;
+    repository.force_push(&format!(
+        "{}/{}:{}",
+        repository.original_remote(),
+        tmp_release_branch.name,
+        pr.branch()
+    ))?;
+
+    Ok(())
+}
+
+/// Temporary branch.
+/// It deletes the branch in remote when it goes out of scope.
+/// In this way, we can ensure that the branch is deleted even if the program panics.
+struct TmpBranch<'a> {
+    name: String,
+    repository: &'a Repo,
+}
+
+impl<'a> TmpBranch<'a> {
+    fn checkout_new(repository: &'a Repo, name: impl Into<String>) -> anyhow::Result<Self> {
+        let name = name.into();
+        repository.checkout_new_branch(&name)?;
+        let branch = Self { name, repository };
+        Ok(branch)
+    }
+}
+
+impl Drop for TmpBranch<'_> {
+    fn drop(&mut self) {
+        if let Err(e) = self.repository.delete_branch_in_remote(&self.name) {
+            tracing::error!("cannot delete branch {}: {:?}", self.name, e);
+        }
```

Comment:
Should we also delete the branch from the local repository?
I know this command is run in a temporary directory, but just to clean up things correctly, in case we reuse this code somewhere else, or if in the future we change the "temporary directory" strategy

Reply:
I was thinking about it. Committing through graphql leaves all the changes uncommitted locally and I wasn't sure if it's okay to remove everything, even untracked files blindly? They can conflict when running `git checkout ...`, and we must switch the branch to delete this temporary one.
Another way would be replacing the `git fetch ...` with `git pull --force ...` so that it would overwrite already committed content making checkout easier.

I decided to leave it like so to not increase complexity however it's definitely an option.

Reply:
> I decided to leave it like so to not increase complexity however it's definitely an option.

Sounds good 👍